apps/wallet/src/ui/app/staking/stake/StakeForm.tsx

```diff
@@ -44,50 +46,109 @@ function StakeForm({
         onClearRef.current();
     }, [amount]);
 
-    const [formatted, symbol] = useFormatCoin(coinBalance, coinType);
+    const [formatted] = useFormatCoin(coinBalance, coinType);
+
+    const [gasBudgetEstimation] = useFormatCoin(
+        DEFAULT_GAS_BUDGET_FOR_STAKE,
+        SUI_TYPE_ARG
+    );
+
+    const coinBalanceMinusGas =
+        coinBalance - BigInt(DEFAULT_GAS_BUDGET_FOR_STAKE);
+
+    const [maxToken, symbol, isLoading] = useFormatCoin(
+        coinBalanceMinusGas,
+        coinType
+    );
+
+    const setMaxToken = useCallback(() => {
+        if (!maxToken) return;
+        setFieldValue('amount', maxToken);
+    }, [maxToken, setFieldValue]);
 
     return (
-        <Form className={st.container} autoComplete="off" noValidate={true}>
-            <div className={st.group}>
-                <label className={st.label}>Amount:</label>
-                <Field
-                    component={NumberInput}
-                    allowNegative={false}
-                    name="amount"
-                    placeholder={`Total ${symbol} to stake`}
-                    className={st.input}
-                    decimals
-                />
-                <div className={st.muted}>
-                    Available balance: {formatted} {symbol}
-                </div>
+        <Form
+            className="flex flex-1 flex-col flex-nowrap"
+            autoComplete="off"
+            noValidate={true}
+        >
+            <Content>
+                <Card
+                    variant="blue"
+                    header={
+                        <div className="p-2.5 w-full flex bg-white">
+                            <Field
+                                component={NumberInput}
+                                allowNegative={false}
+                                name="amount"
+                                className="w-full border-none text-hero-dark text-heading4 font-semibold placeholder:text-gray-70 placeholder:font-medium"
+                                decimals
+                            />
+                            <button
+                                className="bg-white border border-solid border-gray-60 hover:border-steel-dark rounded-2xl h-6 w-11 flex justify-center items-center cursor-pointer text-steel-darker hover:text-steel-darker text-bodySmall font-medium"
+                                onClick={setMaxToken}
+                                disabled={!isLoading}
```

Comment:
this probably should be `isLoading` remove !

Reply:
done